st2common/st2common/models/db/__init__.py

```diff
@@ -97,9 +99,28 @@ def db_ensure_indexes():
     LOG.debug('Ensuring database indexes...')
     model_classes = get_model_classes()
 
-    for cls in model_classes:
-        LOG.debug('Ensuring indexes for model "%s"...' % (cls.__name__))
-        cls.ensure_indexes()
+    for model_class in model_classes:
+        # First clean-up extra indexes
+        cleanup_extra_indexes(model_class=model_class)
+        LOG.debug('Ensuring indexes for model "%s"...' % (model_class.__name__))
+        model_class.ensure_indexes()
+
+
+def cleanup_extra_indexes(model_class):
+    """
+    Finds any extra indexes and removes those from mongodb.
+    """
+    extra_indexes = model_class.compare_indexes().get('extra', None)
```

Comment:
Not sure. I was surprised that this exact method even existed but am glad it does.